Replace all FullWidthLayout with WideLayout

p-jackson opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

## Proposed Changes

* Removes `FullWidthLayout` and replaces it with a `width` parameter on the existing `WideLayout` wireframe
  * The `width` parameter is (mostly) based off the standard gutenberg break points
* Creates a new wireframe for the WIP playground step
* Add some more storybook examples to show the different sizes

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

After discussing here p1743044978661229-slack-C08HKNPL2BG, it looks like we misunderstood what was meant by the "full width" layouts. Actually all the layouts are supposed to have a `max-width`, even the the design picker and design preview steps.

That means we don't need a separate wireframe for "full width" steps, they're the same as the "wide" layout, just with different maximum widths. The maximum widths are based of standard Gutenberg break point widths, with an exception for the plans step, which seems a bit cramped compared to how we have it in production.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Run through `/setup/onboarding` and try resizing the browser on each step.
* This isn't an authoritative max width all steps are supposed to conform to, I went by feel when picking the max width for each step.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

p-jackson committed:
Replace all FullWidthLayout with WideLayout

p-jackson commented on packages/onboarding/src/step-container-v2/wireframes/PlaygroundLayout/PlaygroundLayout.tsx:
```diff
@@ -0,0 +1,32 @@
+	children?: ContentProp;
+}
+
+export const PlaygroundLayout = ( { topBar, className, children }: PlaygroundLayoutProps ) => {
```
@zaerl @zaguiini  I made a special purpose wireframe just for the playground step. It was using the `hasContentPadding` prop to bypass the standard stepper layout patterns. However this prop is really there only as a hack for the Checkout page, we don't want it to become the norm to use this prop.

The playground step diverges so far from the existing stepper patterns that I actually think it warrants a separate wireframe. @zaerl I don't know if this is the final design or whether you're just prototyping? If the playground step ends up looking different (maybe more like the design preview step?) then you might want to switch it back to one of the regular wireframes down the road.

You can see the playground step by navigating to `/setup/onboarding/playground` locally.

p-jackson replied:
![CleanShot 2025-04-01 at 17 47 43@2x](https://github.com/user-attachments/assets/85afb8db-d8c4-4ed1-b5b5-885d25970aaf)

zaerl replied:
Thanks @p-jackson.

> It was using the hasContentPadding prop to bypass the standard stepper layout patterns. 

I didn't know about this. I chose the layout that was giving me the most horizontal space possible, and the `FullWidthLayout` was the best candidate.

> @zaerl I don't know if this is the final design or whether you're just prototyping? If the playground step ends up looking different (maybe more like the design preview step?) then you might want to switch it back to one of the regular wireframes down the road.

This experimental flow is the beginning of integrating Playground in our onboarding flow. This is not the final design, but I'm sure that using 100% of the width of the page will be a must-have. We use an `iframe`, and we need all the space required to preview the page. What can change is the top bar. But it should be some wording or other minimal tweaks.

cc. @akirk @ashfame @bgrgicak if they want to add other info.

p-jackson replied:
Ok, sounds like a custom wireframe is best then 👍

p-jackson commented on packages/onboarding/src/step-container-v2/components/ContentWrapper/ContentWrapper.tsx:
```diff
@@ -3,22 +3,26 @@ import { ReactNode } from 'react';
+type Width =
+	| 'wide'
+	| 'huge'
+	| 'huge-plus' // Non-standard width introduced for plans step
```
This is what the plans step looks like if it uses the standard `huge` breakpoint for the max width

![CleanShot 2025-04-01 at 17 50 41@2x](https://github.com/user-attachments/assets/95d8b9dc-8172-46eb-aadc-68fc3cb1d73e)

This is what it looks like in production. Subtle, but slightly more breathing space.

![CleanShot 2025-04-01 at 17 54 17@2x](https://github.com/user-attachments/assets/051f0692-77aa-4347-bc66-8f67ca69ddbf)

Maybe the inconsistency isn't worth it.

@nuriapenya @fcoveram

zaerl replied:
I tried both. Our six columns need that space. We must consider that some languages are more verbose than English, and the paragraphs are slightly more crowded.

zaguiini replied:
If we're concerned about breathing space for the elements, then we can probably make this layout use the `xhuge` max-width.

It's a big difference, though:
- `huge` uses `1440px`
- `huge-plus` is using `1536px`
- `xhuge` uses `1920px`

I would argue that it's not a bad thing to have it as large as possible for clarity. Here's how it looks:

| `huge-plus` | `xhuge` |
| ----------- | ---------- |
| ![calypso localhost_3000_setup_onboarding_plans (1)](https://github.com/user-attachments/assets/a66dacc6-b77b-4698-abc0-a6b3179980df) | ![calypso localhost_3000_setup_onboarding_plans](https://github.com/user-attachments/assets/7ba94356-1039-41f5-bbf6-f3c55f06ce6f) |

vykes-mac replied:
non-blocking, in terms of naming could be more conventional. maybe `wide` `x-wide` , `2x-wide` or something that the brain might automatically guess the next option. I wouldn't guess `wide` to `huge`

p-jackson replied:
> I wouldn't guess `wide` to `huge`

😅 yeah same. I went with this names to be consistent with the "standard" Gutenberg breakpoints. Core first?

p-jackson commented on packages/onboarding/src/step-container-v2/wireframes/WideLayout/WideLayout.tsx:
```diff
@@ -22,16 +28,23 @@ export const WideLayout = ( {
 	children,
 	footer,
 	stickyBottomBar,
+	width = 'wide',
```
Maybe this prop should actually be called `maxWidth` 🤔

zaerl replied:
> Maybe this prop should actually be called `maxWidth` 🤔

Good idea.

zaguiini replied:
Agreed. I'll change it.

zaerl commented on client/landing/stepper/declarative-flow/internals/steps-repository/unified-plans/unified-plans-step.tsx:
```diff
@@ -495,6 +495,7 @@ function UnifiedPlansStep( {
 			<>
 				<MarketingMessage path="signup/plans" />
 				<Step.WideLayout
+					width="huge-plus"
```
I'm not a big fan of the name used here, but it's perfectly fine to have this.

zaerl reviewed (commented):
I've left some comments, but nothing needs to be changed. This is a great PR and can be landed. But let's wait for the others.

paulopmt1 commented on packages/onboarding/src/step-container-v2/README.md:
```diff
@@ -218,6 +218,15 @@ Of course not! The plan is for this new version to replace `StepContainer` entir
+
+Either:
+
+- `yarn storybook:start` from the root of this package, or
+- `yarn workspace @automattic/onboarding storybook:start` from the root of the monorepo.
```
Thanks for improving the docs! 
TIL about the `yarn workspace @automattic/onboarding storybook:start`

paulopmt1 commented on packages/onboarding/src/step-container-v2/helpers/wireframe-placeholder.tsx:
```diff
@@ -3,7 +3,7 @@ export function WireframePlaceholder( {
 	className,
 }: {
-	height?: number;
+	height?: React.CSSProperties[ 'height' ];
```
TIL about this as well. More context for future readers:

![image](https://github.com/user-attachments/assets/65545e2a-c366-40f7-8fc5-f532f7a98bd9)

paulopmt1 commented on packages/onboarding/src/step-container-v2/wireframes/WideLayout/WideLayout.tsx:
```diff
@@ -13,6 +13,12 @@ interface WideLayoutProps {
 	children?: ContentProp;
 	footer?: ReactNode;
 	stickyBottomBar?: ContentProp;
+	width?: React.ComponentProps< typeof ContentWrapper >[ 'width' ];
```
> It will be removed when checkout no longer needs it.

Thanks for clarifying it!

paulopmt1 commented on client/my-sites/checkout/src/components/checkout-main-content.tsx:
```diff
@@ -830,7 +830,8 @@ export default function CheckoutMainContent( {
 		<StepContainerV2CheckoutFixer isLargeViewport={ isLargeViewport }>
-			<Step.FullWidthLayout
+			<Step.WideLayout
+				width="xhuge"
```
Asked about this `xhuge` variation here: p1743511685985719/1743044978.661229-slack-C08HKNPL2BG
I still prefer the `wide` one:

![image](https://github.com/user-attachments/assets/fd79f28b-9db1-498d-a925-0eaa0a33fcd9)

paulopmt1 reviewed (approved):
Thanks for improving it @p-jackson!

Do we want all screens to have a max-width? Asked here: p1743511685985719/1743044978.661229-slack-C08HKNPL2BG

For instance, if we apply it to the checkout page, it will restrict the background color of the right block on wide monitors:

| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/675c6364-055b-4889-8025-d832e6cdb23c) | ![image](https://github.com/user-attachments/assets/36bfd353-59d5-46dd-8ddd-103cbe3d61b8) |


I liked more the new theme selection implementation. Very nice!

| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/7c86fa66-5d7b-4b6f-8749-caee8029dd93) | ![image](https://github.com/user-attachments/assets/3ab6ea35-8561-4cb7-ac27-6f34c4cfe959) |

zaguiini reviewed (dismissed):
These changes also affect the `TwoColumnLayout` wireframe, which now goes from `1038px` to `1280px` as its maximum width. I asked for confirmation here: p1743530585012359/1743044978.661229-slack-C08HKNPL2BG.

I'm requesting changes so we don't accidentally merge this PR.

zaguiini committed:
Domains' side content uses full width of its column in StepContainerV2's version

zaguiini committed:
Rename prop to maxWidth

zaguiini commented on client/signup/steps/domains/style.scss:
```diff
@@ -546,10 +546,6 @@ body.is-section-signup {
 		flex: 1;
 		width: 100%;
-
-		@include break-large {
-			max-width: 290px;
-		}
```
This was an uncovered bug 🐛 

Because the previous maximum width of the content wrapper was `1038px` in the TwoColumnLayout, the side content wouldn't ever reach more than 290px in width.

But now that it's a bit larger, we were left with the side column taking less space than it should, making it uneven. This patch fixes that.

| Before | After |
| ------- | ----- | 
| ![image](https://github.com/user-attachments/assets/5b28f8ba-a661-498b-97d2-9055c7b2c8fa) | ![image](https://github.com/user-attachments/assets/061e498a-419f-4524-82e3-c129edd8b64e) |

It's a very subtle difference, but not for me 😁

zaguiini committed:
Remove huge-plus

zaguiini committed:
Use CSS grid in ContentWrapper

p-jackson committed:
Do not use column layout on mobile devices

p-jackson committed:
ContentWrapper widths match mockup

The mockups use standard widths for the step content wrapper, however
they don't include the margin at the edge of the screen inside that
total.
This means the goals step heading now wraps the way expected from the
mockup.

p-jackson commented:
The `CenteredContentLayout` wireframe seems broken in the new column system right?

https://github.com/user-attachments/assets/ca5ee5ab-3b1b-4012-9baf-552dba73e0fe

We could switch to mobile layout as soon as the browser window hits the columns max width size. But then that'd be super broken too right?

zaguiini committed:
Use column max width calculation instead of using CSS grid

p-jackson commented on packages/onboarding/src/step-container-v2/components/ContentWrapper/ContentWrapper.tsx:
```diff
@@ -1,29 +1,48 @@
+	return (
+		<div
+			className="step-container-v2__content-wrapper-row"
+			style={ {
+				// @ts-expect-error -- This is a valid CSS variable.
+				'--columns': columns,
+			} }
```
`as` is better in this situation because `@ts-expect-error` is more powerful than we need. It ignores things like spelling mistakes but we don't need something that powerful.

```suggestion
			style={ {
				'--columns': columns,
			} as React.CSSProperties }
```

p-jackson commented on packages/onboarding/src/step-container-v2/components/ContentWrapper/style.scss:
```diff
@@ -6,19 +9,31 @@
+		--step-container-v2-content-max-width: calc(
+			#{$break-xhuge} + 2 * var( --step-container-v2-content-inline-padding )
+		);
+		max-width: var( --step-container-v2-content-max-width );
```
Does the `max-width` rules have to be defined multiple times? Or should it be moved up a level. I'm not a CSS variable expert.

p-jackson reviewed (commented):
- [x] Todo: we still need to implement the fix for the checkout that stretches the grey background all the way to the right edge.

> Let me know what you think of it 🤞🏻

I think it's good.
Ultimately I don't think the designers technically want a column-style layout. And that makes sense; apps these days tend to be truly responsive, not column-based.

But we can still communicate using the language of column layouts, and that's what you've done here.

zaguiini committed:
Cast inline CSS as CSSProperties

zaguiini committed:
Remove leftover grid rule

zaguiini committed:
Remove redundant max-width rules

zaguiini committed:
Fix wide sidebar at checkout

zaguiini committed:
Remove unused footer prop from wireframes

zaguiini committed:
Remove unnecessary wrapper div and rename Content to ContentRow
